server/src/test/java/keywhiz/service/crypto/CryptoModuleTest.java

```diff
@@ -0,0 +1,39 @@
+package keywhiz.service.crypto;
+
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import java.security.SecureRandom;
+import keywhiz.KeywhizConfig;
+import keywhiz.inject.StrictGuiceModule;
+import keywhiz.test.ServiceContext;
+import org.junit.Test;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNotSame;
+
+public class CryptoModuleTest {
+  @Test
+  public void secureRandomIsExplicitlyBound() {
+    KeywhizConfig config = ServiceContext.create().getConfig();
+
+    Injector injector = Guice.createInjector(
+        new CryptoModule(
+            config.getDerivationProviderClass(),
+            config.getContentKeyStore()),
+        new StrictGuiceModule());
+
+    assertNotNull(injector.getInstance(SecureRandom.class));
+  }
+
+  @Test
+  public void secureRandomIsNonSingleton() {
```

Comment:
Is this so that `SecureRandom` starts with a different seed every time?

Reply:
This is actually mostly for documentation. Because we didn't specify any binding for SecureRandom, and because it has a no-argument constructor, Guice was creating a new instance every time. Now, if this new instance was then given to a singleton then it's also effectively a singleton (Guice will never need to create a second instance) but in any case the behavior was implicit. Now, it's explicit, and this test proves that. If we actually wanted the SecureRandom to be a singleton then we'll need to modify the code accordingly, but it was never a singleton to begin with.